app/gui/language/span-tree/src/lib.rs

```diff
@@ -77,19 +77,37 @@ use crate::generate::Context;
 pub struct ArgumentInfo {
     pub name:       Option<String>,
     pub tp:         Option<String>,
+    pub call_id:    Option<ast::Id>,
+    pub target_id:  Option<ast::Id>,
     pub tag_values: Vec<String>,
 }
 
 impl ArgumentInfo {
     /// Constructor.
-    pub fn new(name: Option<String>, tp: Option<String>, tag_values: Vec<String>) -> Self {
-        Self { name, tp, tag_values }
+    pub fn new(
+        name: Option<String>,
+        tp: Option<String>,
+        call_id: Option<ast::Id>,
+        target_id: Option<ast::Id>,
+        tag_values: Vec<String>,
+    ) -> Self {
+        Self { name, tp, call_id, target_id, tag_values }
     }
 
     /// Specialized constructor for "this" argument.
-    pub fn this(tp: Option<String>) -> Self {
+    pub fn this(tp: Option<String>, call_id: Option<ast::Id>) -> Self {
         let name = Some(node::This::NAME.into());
-        Self { name, tp, tag_values: Vec::new() }
+        Self { name, tp, call_id, target_id: None, tag_values: Vec::new() }
+    }
+
+    /// Extend the argument info with the given call id.
+    pub fn with_call_id(self, call_id: Option<ast::Id>) -> Self {
+        Self { call_id, ..self }
+    }
+
+    /// Extend the argument info with the given target id.
+    pub fn with_target_id(self, target_id: Option<ast::Id>) -> Self {
```

Comment:
these docs are not understandable because the "call id" and "target id" were never explained. Please add explanation (somewhere) what exactly are these. It would be cool if this explanation contain an example short code to explain what are these in a real context.